xcodeproj/internal/xcode_targets.bzl

```diff
@@ -856,14 +914,26 @@ def _linker_inputs_to_dto(
     if not linker_inputs:
         return ({}, None)
 
-    if compile_target:
-        self_product_path = compile_target.product.file.path
+    if compile_targets:
+        self_product_paths = [compile_target.product.file.path for compile_target in compile_targets]
     else:
         # Handle `{cc,swift}_{binary,test}` with `srcs` case
-        self_product_path = paths.join(
+        self_product_paths = [paths.join(
             product.package_dir,
             "lib{}.lo".format(name),
-        )
+        )]
```

Comment:
TODO for me: change this to sue `args()` to reduce starlark cpu and memory usage.